app/src/main/java/com/eveningoutpost/dexdrip/Services/Ob1G5CollectionService.java

```diff
@@ -1717,9 +1719,28 @@ public static void processCalibrationState(final CalibrationState state) {
             final PendingIntent pi = PendingIntent.getActivity(xdrip.getAppContext(), G5_SENSOR_STARTED, JoH.getStartActivityIntent(Home.class), PendingIntent.FLAG_UPDATE_CURRENT);
             JoH.showNotification(state.getText(), "Sensor Stopped", pi, G5_SENSOR_STARTED, true, true, false);
             UserError.Log.ueh(TAG, "Native Sensor is now Stopped: " + state.getExtendedText());
+
+            Treatments.SensorStop(null, "Stopped by transmitter");
+
         } else if (is_started && !was_started) {
             JoH.cancelNotification(G5_SENSOR_STARTED);
             UserError.Log.ueh(TAG, "Native Sensor is now Started: " + state.getExtendedText());
+
```

Comment:
I would prefer if we can move this logic in to Treatments so that there is just a simple call from the collection service that specifies the event occurring.

Reply:
I can do this -- will just have to name the methods such that it's obvious that they sometimes don't do anything (e.g., in a state where an event was recently created), since most of the methods in Treatments are direct database row creators/accessors.

Reply:
You can create an additional helper class if you think that makes sense for this. With the logic moved outside of the collector so its just a simple call out on the event occurring then I'd expect to approve the PR.